src/core/raster/qgsrasterdataprovider.h

```diff
@@ -355,10 +355,26 @@ class CORE_EXPORT QgsRasterDataProvider : public QgsDataProvider, public QgsRast
      *         Empty if failed or there are no results (TODO: better error reporting).
      * \note The arbitraryness of the returned document is enforced by WMS standards
      *       up to at least v1.3.0
+     * \see sample(), which is much more efficient for simple "value at point" queries.
      */
-    //virtual QMap<int, QVariant> identify( const QgsPointXY & point, QgsRaster::IdentifyFormat format, const QgsRectangle &extent = QgsRectangle(), int width = 0, int height = 0 );
     virtual QgsRasterIdentifyResult identify( const QgsPointXY &point, QgsRaster::IdentifyFormat format, const QgsRectangle &boundingBox = QgsRectangle(), int width = 0, int height = 0, int dpi = 96 );
 
+    /**
+     * Samples a raster value from the specified \a band found at the \a point position. The context
+     * parameters \a boundingBox, \a width and \a height are important to identify
+     * on the same zoom level as a displayed map and to do effective
+     * caching (WCS). If context params are not specified the highest
+     * resolution is used. capabilities() may be used to test if format
+     * is supported by provider.
+     *
+     * A null QVariant will be returned if the point is outside data source extent, or an invalid
+     * band number was specified.
```

Comment:
Would be good to say what kind of types to expect in QVariant. Looks like currently we only support double... if that's always going to be the case, I would probably prefer to have double as return type + "bool *ok" to indicate success/failure

Reply:
Fixed, I've removed the variant.